codelitephp/PHPParser/FilesCollector.cpp

```diff
@@ -50,11 +51,14 @@ void FilesCollector::Collect(const wxString& rootFolder)
             fullpath << dir.GetNameWithSep() << filename;
             bool isDirectory = wxFileName::DirExists(fullpath);
             if(isDirectory && (m_excludeFolders.count(filename) == 0)) {
-                // A directory
-                Q.push(fullpath);
-                fullpath << wxFileName::GetPathSeparator() << FOLDER_MARKER;
-                V.push_back(fullpath);
-
+                wxString canonicalPath = wxFileName(fullpath).ResolveLink().GetFullPath();
+                if (visitedFolders.find(canonicalPath) == visitedFolders.end()) {
+                    visitedFolders.insert(canonicalPath);
```

Comment:
```suggestion
                if (visitedFolders.insert(canonicalPath).second) {
```
or
```suggestion
                if (const auto [it, inserted] = visitedFolders.insert(canonicalPath); inserted) {
```

To do only one lookup.